docs/proposals/migration/design-of-seamless-cluster-migration-scheme.md

```diff
@@ -0,0 +1,316 @@
+
+- After the migration is completed, the total number of Pod replicas in all clusters remains unchanged.
+
+
+
+`Karmada` currently provides two means to migrate resources from existing clusters to `Karmada`:
+
+1）Method 1:
+
+Call `karmadactl promote deployment xxx -C yyy` to take over the existing deployment and other resources in the original single cluster
+
+2）Method 2: 
+
+Manually write the `PropagationPolicy` and `ResourceTemplate` configurations and call the API interface of `karmada-apiserver` to apply the configuration. 
+
+Noting that **it is necessary to add `work.karmada.io/conflict-resolution: overwrite` in the annotation of `ResourceTemplate`**, indicating that when there is a naming conflict resource in the member cluster, it will be forced taken over by overwriting the configuration **(if left blank, it will not be taken over by default)** 
+
+
+> Detailed practical steps can be found in: [Utilizing existing policy capabilities to achieve cluster migration](https://www.yuque.com/chaosi/pwzt9c/ypugrp6452lu36ok)
+
+
+
+
+The above methods all require **the migration of specific native resources that the operator perceives in the member cluster**. In some `CI/CD` automated pipeline scenarios, **the operator cannot manually modify native resource declarations such as `Deployment`, and the resources contained in the original cluster are also not visible to the operator**. This brings difficulties and challenges to achieving seamless migration.
+
+
+
+### Goals
+
+Explore how to enable `Karmada`  to achieve seamless cluster migration in a more perfect and universal way, to meet more complex user scenarios and achieve complete user insensitivity?
+
+In detail, we are proposed to extend the API semantics of Karmada custom resources to support declaring how to take over a matching resource when there is a conflict in the member cluster.
+
+
+
+### Non-Goals
+
+If there is a conflict in the member cluster and the resource is not taken over during the conflict, how to ensure that the total number of replicas meets the expected `ResourceTemplate`? This follows the original logic and is not within the scope of this proposal discussion.
+
+> For example, the `ResourceTemplate` specifies that two replicas need to be evenly allocated to two member clusters. If member cluster 1 fails to allocate due to conflicts and does not take over, the proposal does not care about how to handle the failure issue.
+
+
+
+## Proposal
+
+1. For custom resources such as `PropagationPolicy`, `ResourceBinding`, and `Work`, add a field called `conflictResolution`. The operator can use this field in the `PropagationPolicy` to specify how to take over the matched resource when there is a conflict in the member cluster.
+
+2. The field `conflictResolution` has the following values:
+
+   * `overwrite` : continue to propogate the resource and overwrite the new configuration to member clusters
```

Comment:
how to define `overwrite`? force update? but some filelds of some resources are immutable, even force update you will get an error, like `pod.spec.nodeName` etc. maybe you can only delete and recreate them. so I think you should clearly state in the documentation about how it works and what's its restriction.